### PR TITLE
refactor(metrics): move reward into diffusion_planner.metrics

### PR DESCRIPTION
**Pull request by go-sakayori**
## PR1 for #130 — pure move of subscores/geometry/config into `diffusion_planner.metrics`

Carves the raw EPDMS-style subscore suite out of `rlvr/reward.py` into a neutral `diffusion_planner.metrics` package so the base-SFT validation loop (`valid_predictor.py`) can import the same subscores RLVR optimizes, **without depending on `rlvr`**. This is the **"move" half** — strictly behavior-preserving. The behavior-preserving *split* of `compute_reward_batch` into `compute_subscores_batch` + `_shape_reward` is PR2.

### What moved (byte-identical)
```
diffusion_planner/diffusion_planner/metrics/
  config.py      RewardConfig
  geometry.py    ego bbox, point-in-polygon, point↔segment distance,
                 _closest_points_between_rects, _build_lane_polygons (both variants),
                 _build_sg_diff_kernel, _classify_outer_boundaries, ...
  subscores.py   compute_{safety,ttc,feasibility,kinematic_gate,centerline,progress,
                 smoothness,red_light}_score_batch, compute_road_border_penalty,
                 compute_static_collision_penalty, compute_lane_departure_penalty
                 (+ their constants and shared Savitzky–Golay kernel caches)
  __init__.py    re-exports the moved API
```

### What stayed in `rlvr/reward.py`
Reward **shaping** only: `RewardBreakdown`, `compute_reward_batch`, `compute_reward`, `compute_group_advantages`. `reward.py` shrinks 3386 → 639 lines and becomes a re-export shim:

```python
from diffusion_planner.metrics.config import *
from diffusion_planner.metrics.geometry import *
from diffusion_planner.metrics.subscores import *
```

so `from rlvr.reward import ...` keeps working unchanged for every existing caller (≈30 modules).

### Notes / decisions
- **Pure move, no logic change.** Function/class bodies are byte-identical (verified by ast source comparison). The two same-named `_build_lane_polygons` (a dead list-returning v1 shadowed by the edge-returning v2) and the currently-unused `_LN_*` constants are preserved exactly to keep behavior identical.
- `metrics/` imports only `math` / `numpy` / `torch` / `diffusion_planner.model.guidance.collision` — **no `rlvr` dependency** (the design requirement).
- No packaging change: `metrics` is importable like the existing `diffusion_planner.model` / `diffusion_planner.utils` subpackages under the editable/workspace install.

### Verification

```
### 1. byte-identical move + re-export check
moved def/class sources byte-identical: True  (31 defs)
symbols imported from rlvr.reward across repo: 27  | missing re-exports: NONE

### 2. golden regression (compute_reward_batch behavior pinned by PR0)
16 passed, 1 warning in 3.81s

### 3. all test files that import rlvr.reward
123 passed, 4 warnings in 4.82s
files: test_lat_accel test_exploration_trainer test_progress_penalty_bugfix
       test_generation_variants test_static_collision test_reward
       test_reward_gates test_reward_golden
```

The PR0 golden tests (`rlvr/test_reward_golden.py`) prove `compute_reward_batch` output is unchanged by the move. Run locally with:
```
PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 uv run python -m pytest rlvr/test_reward_golden.py
```
(CI currently runs `pre-commit` only — no pytest workflow — so the suite above was run locally.)

### Next
- **PR2** — behavior-preserving split `compute_reward_batch → compute_subscores_batch (raw) + _shape_reward (weights/gates)`; introduce `SubscoreBreakdown`.
- **PR3** — call `compute_subscores_batch` in `validate_model`, log `valid_loss/epdms/*`.

### CODE REVIEW

File: diffusion_planner/diffusion_planner/metrics/geometry.py [hunk 1]
```diff
@@ -0,0 +1,948 @@
...
+
+import math
+
+import numpy as np
+import torch
+
+from diffusion_planner.model.guidance.collision import (
+    batch_signed_distance_rect,
+    center_rect_to_points,
+)
```
[the remark this answers is not in the document]
Comment (go-sakayori) [reply diffusion_planner/diffusion_planner/metrics/geometry.py]: Fixed 842a3cb9a733a81e4ed509fbecfcba106f9bf7cf

File: diffusion_planner/diffusion_planner/metrics/geometry.py [hunk 2]
```diff
@@ -0,0 +1,948 @@
...
+    # If best < 16, q is on rect1, foot is on rect2. Else swap.
+    is_on_r1 = best < 16
+    pt1 = torch.where(is_on_r1.unsqueeze(-1), q_best, foot_best)
+    pt2 = torch.where(is_on_r1.unsqueeze(-1), foot_best, q_best)
+    return pt1.to(dtype), pt2.to(dtype)
+
+
+def _build_lane_polygons(
+    lanes: torch.Tensor,
+) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, int]:
```
[the remark this answers is not in the document]
Comment (go-sakayori) [reply diffusion_planner/diffusion_planner/metrics/geometry.py:497]: Fixed 842a3cb9a733a81e4ed509fbecfcba106f9bf7cf

File: rlvr/reward.py [hunk 3]
```diff
@@ -12,2797 +12,55 @@
...
-    else:
-        cont_penalty = torch.where(
-            is_in_ts,
-            (1.0 - per_ts_min[:, 1:] / lane_cont_thresh).clamp(min=0, max=1),
-            torch.zeros_like(per_ts_min[:, 1:]),
-        ).mean(dim=1)
-
-    return crossing_gate, near_frac, wide_frac, lane_crossing_steps, cont_penalty
+from diffusion_planner.metrics.config import *  # noqa: F401,F403
+from diffusion_planner.metrics.config import RewardConfig  # noqa: F401
```
[the remark this answers is not in the document]
Comment (go-sakayori) [reply rlvr/reward.py]: Fixed 842a3cb9a733a81e4ed509fbecfcba106f9bf7cf

File: diffusion_planner/diffusion_planner/metrics/__init__.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+Raw EPDMS-style subscores + geometry + config, carved out of ``rlvr.reward`` so
+the base-SFT validation loop can import them without depending on ``rlvr``.
+``rlvr.reward`` re-exports the same symbols for backward compatibility.
+"""
+
+from __future__ import annotations
+
+from diffusion_planner.metrics.config import *  # noqa: F401,F403
+from diffusion_planner.metrics.config import RewardConfig  # noqa: F401
```
[the remark this answers is not in the document]
Comment (go-sakayori) [reply diffusion_planner/diffusion_planner/metrics/__init__.py]: Fixed 842a3cb9a733a81e4ed509fbecfcba106f9bf7cf
